src/components/MoneyRequestConfirmationList.js

```diff
@@ -125,18 +151,41 @@ const defaultProps = {
     receiptSource: '',
     listStyles: [],
     policyCategories: {},
+    transactionID: '',
+    transaction: {},
+    mileageRate: {unit: CONST.CUSTOM_UNITS.DISTANCE_UNIT_MILES, rate: 0, currency: 'USD'},
+    isDistanceRequest: false,
 };
 
 function MoneyRequestConfirmationList(props) {
     // Destructure functions from props to pass it as a dependecy to useCallback/useMemo hooks.
     // Prop functions pass props itself as a "this" value to the function which means they change every time props change.
-    const {onSendMoney, onConfirm, onSelectParticipant} = props;
+    const {onSendMoney, onConfirm, onSelectParticipant, transaction} = props;
     const {translate} = useLocalize();
 
     // A flag and a toggler for showing the rest of the form fields
     const [showAllFields, toggleShowAllFields] = useReducer((state) => !state, false);
     const isTypeRequest = props.iouType === CONST.IOU.MONEY_REQUEST_TYPE.REQUEST;
 
+    const {unit, rate, currency} = props.mileageRate;
+    const distance = lodashGet(transaction, 'routes.route0.distance', 0);
+    const shouldCalculateDistanceAmount = props.isDistanceRequest && props.iouAmount === 0;
+    const shouldCategoryEditable = !_.isEmpty(props.policyCategories) && !props.isDistanceRequest;
+
+    const formattedAmount = CurrencyUtils.convertToDisplayString(
+        shouldCalculateDistanceAmount ? DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate) : props.iouAmount,
+        props.isDistanceRequest ? currency : props.iouCurrencyCode,
+    );
+
+    useEffect(() => {
+        if (!shouldCalculateDistanceAmount) {
+            return;
+        }
+
+        const amount = DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate);
+        IOU.setMoneyRequestAmount(amount);
```

Comment:
Coming from https://github.com/Expensify/App/issues/26946.

I'm trying to figure out this `props.iouAmount === 0` -> `setMoneyRequestAmount` -> `props.iouAmount !== 0` cycle to review the proposals properly.

Would it be fine to drop the `props.iouAmount === 0` condition and do something like:

```js
// useMemo if we consider this heavy
const distanceAmmount = props.isDistanceRequest ? DistanceRequestUtils.getDistanceRequestAmount(distance, unit, rate) : undefined;

const formattedAmount = distanceAmmount !== undefined ?
    CurrencyUtils.convertToDisplayString(distanceAmmount, currency) :
    CurrencyUtils.convertToDisplayString(props.iouAmount, props.iouCurrencyCode);

useEffect(() => {
    if (distanceAmmount !== undefined) {
        IOU.setMoneyRequestAmount(distanceAmmount);
    }
}, [distanceAmmount]);
```

...or would it mess with the original intention/semantics of the `props.iouAmount === 0` check, in the way I'm missing?

Reply:
(Continuing [here](https://github.com/Expensify/App/pull/25707/files#r1322653311))